### PR TITLE
Add checkbox to search for all the branches by commit message

### PR DESCRIPTION
**Pull request by gzsombor**
This patch adds the ability to search for all the commits containing the entered text in their commit message, not only in the current branch

### CODE REVIEW

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -513,6 +513,7 @@ editor.upload_files_to_dir = Upload files to '%s'
 
 commits.commits = Commits
 commits.search = Search commits
+commits.searchAll = All
```
Comment (andreynering) [inline options/locale/locale_en-US.ini]: Use snake_case:

`searchAll` -> `search_all`

File: vendor/code.gitea.io/git/commit.go [hunk 2]
```diff
@@ -198,8 +198,8 @@ func (c *Commit) CommitsBeforeUntil(commitID string) (*list.List, error) {
 }
 
 // SearchCommits returns the commits match the keyword before current revision
-func (c *Commit) SearchCommits(keyword string) (*list.List, error) {
-	return c.repo.searchCommits(c.ID, keyword)
+func (c *Commit) SearchCommits(keyword string, all bool) (*list.List, error) {
+	return c.repo.searchCommits(c.ID, keyword, all)
```
Comment (andreynering) [inline vendor/code.gitea.io/git/commit.go:202]: You should not change this package directly in the vendor folder.

You should send PR to https://github.com/go-gitea/git first, and after aproved use [govendor][govendor] to update the changes here.

[govendor]: https://github.com/kardianos/govendor

File: templates/repo/commits_table.tmpl [hunk 3]
```diff
@@ -6,6 +6,7 @@
 				<div class="ui tiny search input">
 					<input name="q" placeholder="{{.i18n.Tr "repo.commits.search"}}" value="{{.Keyword}}" autofocus>
 				</div>
+				<input type="checkbox" name="all" id="all" value="true" {{.All}}><label for="all">{{.i18n.Tr "repo.commits.searchAll"}}</label>
```
Comment (andreynering) [inline templates/repo/commits_table.tmpl]: Here, too:

`searchAll` -> `search_all`

**Review by andreynering — CHANGES_REQUESTED** [review]
It's working, but need few changes.

**Comment by gzsombor** [comment]
Hope, I've done it correctly, I've never used go before :)

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Comment by lunny** [comment]
@andreynering please confirm.

**Review by andreynering — APPROVED** [review]

**Comment by andreynering** [comment]
@lunny Approved

@gzsombor Thank you for your PRs and fast changes.

**Comment by lunny** [comment]
@andreynering then change this to v1.1 or wait to v1.2?

**Comment by andreynering** [comment]
@lunny I think this change is simple enough and can be merged now.

**Comment by andreynering** [comment]
LGTM

**Pull request merged**